packages/indy-vdr/src/pool/IndyVdrPoolService.ts

```diff
@@ -0,0 +1,230 @@
+import type { AgentContext } from '../../../core/src/agent'
+import { GetNymRequest, indyVdr,} from 'indy-vdr-test-shared'
+
+import {
+  AgentDependencies,
+  Logger,
+  InjectionSymbols,
+  injectable,
+  inject,
+  LedgerError,
+  LedgerNotConfiguredError,
+  LedgerNotFoundError,
+} from '@aries-framework/core'
+
+import { IndyVdrError, GetNymResponse } from 'indy-vdr-test-shared'
+import { CacheRepository, PersistedLruCache } from '../../../core/src/cache'
+import { IndySdkError } from '../../../core/src/error'
+import { isSelfCertifiedDid } from '../../../core/src/utils/did'
+import { isIndyError } from '../../../core/src/utils/indyError'
+import { allSettled, onlyFulfilled, onlyRejected } from '../../../core/src/utils/promises'
+import { assertIndyWallet } from '../../../core/src/wallet/util/assertIndyWallet'
+import { IndyVdrPool } from './IndyVdrPool'
+
+export const DID_POOL_CACHE_ID = 'DID_POOL_CACHE'
+export const DID_POOL_CACHE_LIMIT = 500
+export interface CachedDidResponse {
+  nymResponse: {
+    did: string
+    verkey: string
+  }
+  poolId?: string
+}
+@injectable()
+export class IndyVdrPoolService {
+  public pools: IndyVdrPool[] = []
+  private logger: Logger
+  private indyVdr!: typeof indyVdr
+  private agentDependencies: AgentDependencies
+  private didCache: PersistedLruCache<CachedDidResponse>
+
+  public constructor(
+    cacheRepository: CacheRepository,
+    @inject(InjectionSymbols.AgentDependencies) agentDependencies: AgentDependencies,
+    @inject(InjectionSymbols.Logger) logger: Logger
+  ) {
+    this.logger = logger
+    this.agentDependencies = agentDependencies
+
+    this.didCache = new PersistedLruCache(DID_POOL_CACHE_ID, DID_POOL_CACHE_LIMIT, cacheRepository)
+  }
+
+  /**
+   * Create connections to all ledger pools
+   */
+  public async connectToPools() {
+    const handleArray: number[] = []
+    // Sequentially connect to pools so we don't use up too many resources connecting in parallel
+    for (const pool of this.pools) {
+      this.logger.debug(`Connecting to pool: ${pool.id}`)
+      const poolHandle = await pool.connect()
+      this.logger.debug(`Finished connection to pool: ${pool.id}`)
+      handleArray.push(poolHandle)
+    }
+    return handleArray
+  }
+
+  /**
+   * Get the most appropriate pool for the given did. The algorithm is based on the approach as described in this document:
+   * https://docs.google.com/document/d/109C_eMsuZnTnYe2OAd02jAts1vC4axwEKIq7_4dnNVA/edit
+   */
+  public async getPoolForDid(agentContext: AgentContext, did: string): Promise<{ pool: IndyVdrPool }> {
+    // Check if the did starts with did:indy
+    if (did.startsWith('did:indy')) {
+      const nameSpace = did.split(':')[2]
+
+      const pool = this.pools.find((pool) => pool.IndyNamespace === nameSpace)
+
+      if (pool) return { pool }
```

Comment:
I think we can return the pool on the top level. So just return pool instead of return { pool }